src/Microsoft.AspNet.Diagnostics/StatusCodePagesMiddleware.cs

```diff
@@ -25,8 +25,17 @@ public StatusCodePagesMiddleware(RequestDelegate next, [NotNull]StatusCodePagesO
 
         public async Task Invoke(HttpContext context)
         {
+            context.SetFeature<IStatusCodeFeature>(new StatusCodeFeature());
+
             await _next(context);
 
+            var statusCodeFeature = context.GetFeature<IStatusCodeFeature>();
+            if (statusCodeFeature != null && !statusCodeFeature.Enabled)
```

Comment:
Most of this ambiguity is coming from the fact that you put something in the feature collection and then check if it's still there just a few lines later. This assigns redundant meaning to it's removal.

I think this impl would make more sense if you stored the feature object in a local variable and just rely on the value of `Enabled`.